pom.xml

```diff
@@ -457,7 +458,7 @@ under the License.
                     <dependency>
                         <groupId>org.apache.flink</groupId>
                         <artifactId>flink-ci-tools</artifactId>
-                        <version>${flink.version}</version>
+                        <version>${flink-ci-tools.version}</version>
```

Comment:
I think we can just change the `flink.version` to `1.18-SNAPSHOT`

Reply:
Let me think of it. If we accept this change, https://github.com/apache/flink-connector-pulsar/pull/54#discussion_r1265051962 can be applied also.

But before we cut a release, we should always pin a non-snapshot version.

Reply:
The downside is that main branch and PRs may be suddenly broken due to SNAPSHOT changes, but it _should_ not happen and we sooner or later need to fix it.

Reply:
> Let me think of it. If we accept this change, [#54 (comment)](https://github.com/apache/flink-connector-pulsar/pull/54#discussion_r1265051962) can be applied also.
> 
> But before we cut a release, we should always pin a non-snapshot version.

In my opinion, we should checkout a 4.0 branch for the 1.7 Flink.

Reply:
There is already one https://github.com/apache/flink-connector-pulsar/tree/v4.0.

Do you mean we move to 1.18 on master and cut a new release when 1.18 release and for 1.18+ only?

Although, these breaking changes existing in test code only so it should be still compatible with 1.17.

But I'd prefer to do this in another round of PR. This PR focus on upgrading Pulsar version and these changes are workaround for the current failing tests.

Reply:
I see. Bump the master branch with Flink 1.17 and Pulsar 3.0. Then we merge back to 4.0 branch and change the master branch to Flink 1.18 snapshort. Am I right?